crates/utils/src/utils/validation.rs

```diff
@@ -319,6 +336,12 @@ mod tests {
     let actor_name_max_length = 20;
     assert!(is_valid_display_name("hello @there", actor_name_max_length).is_ok());
     assert!(is_valid_display_name("@hello there", actor_name_max_length).is_err());
+    assert!(is_valid_display_name("\u{200d}hello", actor_name_max_length).is_err());
+    assert!(is_valid_display_name(
+      "\u{1f3f3}\u{fe0f}\u{200d}\u{26a7}\u{fe0f}",
+      actor_name_max_length
+    )
+    .is_ok());
```

Comment:
Add another test case where `has_3_permitted_display_chars()` returns false, to ensure that it works correctly.